src/intrusive_pointer.rs

```diff
@@ -99,6 +99,38 @@ unsafe impl<T: ?Sized> IntrusivePointer<T> for Arc<T> {
     }
 }
 
+/// Creates an `IntrusivePointer` from a raw pointer
+///
+/// This method is only safe to call if the raw pointer is known to be
+/// managed by the provided `IntrusivePointer` type.
+pub unsafe fn clone_pointer_from_raw<P: IntrusivePointer<T> + Clone, T: ?Sized>(
```

Comment:
Could you make this helper function `pub(crate)`?

Reply:
done